tests/conftest.py

```diff
@@ -34,6 +31,26 @@ def initial_providers(self) -> List[ConfigProvider]:
 ConfigProvidersConfiguration.enable_airflow_secrets = False
 ConfigProvidersConfiguration.enable_google_secrets = False
 
+CACHED_GOOGLE_SECRETS: Dict[str, Any] = {}
+
+
+class CachedGoogleSecretsProvider(GoogleSecretsProvider):
+    def _look_vault(self, full_key, hint):
+        if full_key not in CACHED_GOOGLE_SECRETS:
+            CACHED_GOOGLE_SECRETS[full_key] = super()._look_vault(full_key, hint)
+        return CACHED_GOOGLE_SECRETS[full_key]
+
+    def _list_vault(self):
+        key_ = "__list_vault"
+        if key_ not in CACHED_GOOGLE_SECRETS:
+            CACHED_GOOGLE_SECRETS[key_] = super()._list_vault()
+        return CACHED_GOOGLE_SECRETS[key_]
+
+
+from dlt.common.configuration.providers import google_secrets
+
+google_secrets.GoogleSecretsProvider = CachedGoogleSecretsProvider  # type: ignore[misc]
```

Comment:
I don't think this will work this way. I tried something like that when I was testing and it did not work, and I just checked out your branch and the cachedprovider is not used unfortunately..